apps/opik-backend/src/main/java/com/comet/opik/infrastructure/bi/InstallationReportService.java

```diff
@@ -0,0 +1,135 @@
+import jakarta.ws.rs.client.Client;
+import jakarta.ws.rs.client.Entity;
+import jakarta.ws.rs.core.MediaType;
+import jakarta.ws.rs.core.Response;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import reactor.core.publisher.Mono;
+import reactor.core.scheduler.Schedulers;
+import ru.vyarus.dropwizard.guice.module.lifecycle.GuiceyLifecycle;
+
+import java.net.URI;
+import java.util.Map;
+import java.util.Optional;
+
+@ImplementedBy(InstallationReportServiceImpl.class)
+interface InstallationReportService {
+    String NOTIFICATION_EVENT_TYPE = "opik_os_startup_be";
+
+    void reportInstallation();
+}
+
+@Singleton
+@Slf4j
+@RequiredArgsConstructor(onConstructor_ = @Inject)
+class InstallationReportServiceImpl implements InstallationReportService {
+
+    private final LockService lockService;
+    private final IdGenerator generator;
+    private final UsageReportService usageReport;
+    private final OpikConfiguration config;
+    private final Client client;
+
+    public void reportInstallation() {
+
+        String eventType = GuiceyLifecycle.ApplicationStarted.name();
+
+        if (!config.getUsageReport().isEnabled()) {
+            log.info("Usage report is disabled");
+            return;
+        }
+
+        if (StringUtils.isEmpty(config.getUsageReport().getUrl())) {
+            log.warn("Usage report URL is not set");
+            return;
+        }
+
+        var lock = new LockService.Lock("opik-%s".formatted(eventType));
+
+        lockService.executeWithLock(lock, tryToReportStartupEvent(eventType))
+                .subscribeOn(Schedulers.boundedElastic())
+                .onErrorResume(e -> {
+                    log.warn("Didn't reported due to error", e);
+                    return Mono.empty();
+                }).block();
+
+    }
+
+    private Mono<Void> tryToReportStartupEvent(String eventType) {
+        return Mono.fromCallable(() -> {
+
+            String anonymousId = getAnonymousId();
+
+            log.info("Anonymous ID: {}", anonymousId);
+
+            if (usageReport.isEventReported(eventType)) {
+                log.info("Event already reported");
+                return null;
+            }
+
+            reportEvent(anonymousId, eventType);
+            return null;
+        });
+    }
+
+    private String getAnonymousId() {
+        var anonymousId = usageReport.getAnonymousId();
+
+        if (anonymousId.isEmpty()) {
+            log.info("Anonymous ID not found, generating a new one");
+            var newId = generator.generateId();
+            log.info("Generated new ID: {}", newId);
+
+            // Save the new ID
+            usageReport.saveAnonymousId(newId.toString());
+
+            anonymousId = Optional.of(newId.toString());
+        }
+
+        return anonymousId.get();
+    }
+
+    private void reportEvent(String anonymousId, String eventType) {
+
+        usageReport.addEvent(eventType);
```

Comment:
As mentioned in my previous comment, not sure what's the point of recording an event that has not yet happened, specially when you're guarding this as a critical section with a Redis distributed lock.

The only use case is to track failed attempts of sending the event.